src/metabase/lib/schema/util.cljc

```diff
@@ -84,21 +83,28 @@
 (mu/defn mbql-clause-distinct-key
   "For deduplicating MBQL clauses: keep just the keys in options that are essential to distinguish one clause from
   another. Removes namespaced keywords and type information keys like `:base-type`."
-  [[tag opts & children]]
-  (into [tag
-         (opts-distinct-key opts)]
-        (map (fn [child]
-               (cond-> child
-                 (mbql-clause? child) mbql-clause-distinct-key)))
-        children))
+  [[tag opts first-child :as clause]]
```

Comment:
This looks like a different version of the change in #63470, I sorta preferred that version honestly since it's a little more concise